internal/provider/kubernetes/gateway.go

```diff
@@ -67,12 +67,26 @@ func newGatewayController(mgr manager.Manager, cfg *config.Server, su status.Upd
 	if err := c.Watch(
 		&source.Kind{Type: &gwapiv1b1.Gateway{}},
 		&handler.EnqueueRequestForObject{},
-		predicate.NewPredicateFuncs(r.hasMatchingController),
+		predicate.NewPredicateFuncs(r.gatewayHasMatchingController),
 	); err != nil {
 		return err
 	}
 	r.log.Info("watching gateway objects")
 
+	// Only enqueue GatewayClass objects that match this Envoy Gateway's controller name.
+	if err := c.Watch(
+		&source.Kind{Type: &gwapiv1b1.GatewayClass{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.NewPredicateFuncs(r.classHasMatchingController),
+	); err != nil {
+		return err
+	}
+	r.log.Info("watching gatewayclass objects")
+
+	// Trigger Gateway reconciliation when a managed GatewayClass has changed.
+	if err := c.Watch(&source.Kind{Type: &gwapiv1b1.GatewayClass{}}, r.enqueueRequestForOwningGatewayClass()); err != nil {
+		return err
+	}
```

Comment:
Can you explain why this is needed in addition to the above watch on GatewayClasses? I'd think the first one would cover it (provided the comment is addressed).